fs/inode/fs_inodesearch.c

```diff
@@ -430,6 +430,29 @@ static int _inode_search(FAR struct inode_search_s *desc)
   return ret;
 }
 
+/****************************************************************************
+ * Name: _inode_getcwd
+ *
+ * Description:
+ *   Return the current working directory
+ *
+ ****************************************************************************/
+
+static FAR const char *_inode_getcwd(void)
+{
+  FAR const char *pwd = "";
+
+#ifndef CONFIG_DISABLE_ENVIRON
+  pwd = getenv("PWD");
+  if (pwd == NULL)
+    {
+      pwd = CONFIG_LIB_HOMEDIR;
+    }
+#endif
+
+  return pwd;
+}
+
```

Comment:
If feels awkward to use CONFIG_LIB_HOMEDIR since that is a setting for the C library.  I am not recommending any change, but it does not feel right to me.

Reply:
This logic will not work if PWD is not defined on the thread.  But I can find only one case where this will be an issue:

    fs/mount/fs_automount.c:  ret = inode_search(&desc);

Reply:
If both PWD and CONFIG_LIB_HOMEDIR doesn't exist or are empty, the follow code:
```
asprintf(&desc->buffer, "%s/%s", _inode_getcwd(), desc->path);
```
will add the / to the relative path which mean the root is the current directory in this case.

Reply:
> getenv() is prototyped in stdlib.h which is not included by this file. It may be being included through some sneak patch, but it is safer if stdlib.h were included explicitly.

Include stdlib.h now.